pyeudiw/federation/static_trust_chain_builder.py

```diff
@@ -0,0 +1,74 @@
+from cryptojwt.jwk.ec import new_ec_key
+from cryptojwt.jws.jws import JWS
+from dataclasses import dataclass
+from pyeudiw.tools.utils import iat_now, exp_from_now
+    
+@dataclass
+class TrustedAnchorInfo:
+    exp_in: int
+    issuer: str
+    subject: str
+    trust_marks: list
+
+@dataclass
+class LeafInfo(TrustedAnchorInfo):
+    metadata: dict
+    authority_hints: list
+    
+@dataclass
+class IntermediateInfo(TrustedAnchorInfo):
+    metadata_policy: dict
+
+def _gen_empty_jwks_field():
+    return {"keys": []}
+
+def _gen_ec(node: LeafInfo | IntermediateInfo | TrustedAnchorInfo, keys: list) -> dict:
+    ec = {
+		'exp': exp_from_now(node.exp_in), # TODO: get all the timestamp dinamically
+		'iat': iat_now(),
+		'iss': node.issuer,
+		'sub': node.subject,
+		'jwks': _gen_empty_jwks_field(), # UPDATED LATER IN THE CODE
+		'trust_marks': node.trust_marks,
+	}    
+    
+    is_leaf = isinstance(node, LeafInfo)
+    
+    if is_leaf:
+        ec["metadata"] = node.metadata
+        ec["authority_hints"] = node.authority_hints
+        ec["metadata"]["openid_relying_party"] = {"jwks": _gen_empty_jwks_field()}
+    elif isinstance(node, IntermediateInfo):
+        ec["metadata_policy"] = node.metadata_policy
+    
+    if not isinstance(keys, list):
+        keys = [keys]
+        
+    for key in keys:
+        serialized_key = key.serialize()
+        ec["jwks"]['keys'].append(serialized_key)
+        
+        if is_leaf:
+            ec['metadata']['openid_relying_party']["jwks"]['keys'].append(serialized_key)
+    
+    return ec
+                
+def _sign_ec(ec: dict, jwk: dict) -> str:
+    signer = JWS(ec, alg="ES256", typ="application/entity-statement+jwt")
```

Comment:
per la firma della trust chain usare sempre RS256 che è il recommended nella specifica federation 1.0

Reply:
Resolved in [fix: changed signature alg](https://github.com/italia/eudi-wallet-it-python/pull/74/commits/3b7b7ffa9dc27f116478c526ece57b04189f9e74)